### ISSUE TITLE
Random Channel Crash

### ISSUE THREAD

**Issue by iAlpex** [ronancpl/HeavenMS#203]
After starting the server and it running for a couple hours or so the server will still show online but when attempting to log in>select world/channel>click character it will kick you back to login screen with an error. (shown below) 
![stuck login](https://user-images.githubusercontent.com/35554049/40815146-f76cd748-6509-11e8-93c5-ddc0a4489548.png)
Please HELP!

you can log in on a different character or channel but eventually the same thing happens.

**Comment by Arufonsu** [comment]
i have the exact issue with java 7,difference is that i cant press start game at all with any character. It goes back to login screen sqying "that id is already logged in" 
somehow the issue is not a problem when working on with java 8, ( but then you find yourself with lack of compatibility for most of the event scripts... x.x) so i highly doubt is a problem with the source, we probably just have something wrong with the java enviorments... have u tried to install jre7 files + jdk7 files + jce7 files? im trying that out tonight when i get home!

**Comment by xinyifly** [comment]
I met similar problem, related logs here:

```
Log: usedCommands.txt
Human used: !sp on 11-07-2018 05:05

Log: usedCommands.txt
Human used: @commands on 11-07-2018 05:05

Log: usedCommands.txt
Human used: !empowerme on 11-07-2018 05:05

Log: usedCommands.txt
Human used: !empowerme on 11-07-2018 05:05

Log: usedCommands.txt
Human used: @commands on 11-07-2018 05:05

Log: sessions.txt
IoSession with /111.19.92.156:21479 opened on 11-07-2018 05:06
Log: sessions.txt
IoSession with /119.28.178.241:52662 opened on 11-07-2018 05:06
Log: sessions.txt
IoSession with /111.19.92.156:10441 opened on 11-07-2018 05:07
Log: sessions.txt
IoSession with /111.19.92.156:22491 opened on 11-07-2018 05:16
Log: sessions.txt
IoSession with /111.19.92.156:22493 opened on 11-07-2018 05:16
disconnect by exception
java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:384)
	at org.apache.mina.transport.socket.nio.NioProcessor.read(NioProcessor.java:273)
	at org.apache.mina.transport.socket.nio.NioProcessor.read(NioProcessor.java:44)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.read(AbstractPollingIoProcessor.java:690)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.process(AbstractPollingIoProcessor.java:664)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.process(AbstractPollingIoProcessor.java:653)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.access$600(AbstractPollingIoProcessor.java:67)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor$Processor.run(AbstractPollingIoProcessor.java:1124)
	at org.apache.mina.util.NamePreservingRunnable.run(NamePreservingRunnable.java:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
	at java.lang.Thread.run(Thread.java:748)
Log: sessions.txt
IoSession with /111.19.92.156:22475 opened on 11-07-2018 05:16
disconnect by exception
java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:384)
	at org.apache.mina.transport.socket.nio.NioProcessor.read(NioProcessor.java:273)
	at org.apache.mina.transport.socket.nio.NioProcessor.read(NioProcessor.java:44)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.read(AbstractPollingIoProcessor.java:690)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.process(AbstractPollingIoProcessor.java:664)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.process(AbstractPollingIoProcessor.java:653)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.access$600(AbstractPollingIoProcessor.java:67)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor$Processor.run(AbstractPollingIoProcessor.java:1124)
	at org.apache.mina.util.NamePreservingRunnable.run(NamePreservingRunnable.java:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
	at java.lang.Thread.run(Thread.java:748)
disconnect by exception
java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:384)
	at org.apache.mina.transport.socket.nio.NioProcessor.read(NioProcessor.java:273)
	at org.apache.mina.transport.socket.nio.NioProcessor.read(NioProcessor.java:44)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.read(AbstractPollingIoProcessor.java:690)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.process(AbstractPollingIoProcessor.java:664)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.process(AbstractPollingIoProcessor.java:653)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.access$600(AbstractPollingIoProcessor.java:67)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor$Processor.run(AbstractPollingIoProcessor.java:1124)
	at org.apache.mina.util.NamePreservingRunnable.run(NamePreservingRunnable.java:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
	at java.lang.Thread.run(Thread.java:748)
Log: sessions.txt
IoSession with /111.19.92.156:22516 opened on 11-07-2018 05:17
Log: sessions.txt
IoSession with /111.19.92.156:22522 opened on 11-07-2018 05:17
Log: sessions.txt
IoSession with /111.19.92.156:10457 opened on 11-07-2018 05:18
Log: sessions.txt
IoSession with /111.19.92.156:10489 opened on 11-07-2018 05:18
Log: sessions.txt
IoSession with /111.19.92.156:10455 opened on 11-07-2018 05:20
Log: sessions.txt
IoSession with /111.19.92.156:21489 opened on 11-07-2018 05:22
Log: sessions.txt
IoSession with /111.19.92.156:10455 opened on 11-07-2018 05:26
Log: sessions.txt
IoSession with /111.19.92.156:10463 opened on 11-07-2018 05:26
Log: sessions.txt
IoSession with /111.19.92.156:10494 opened on 11-07-2018 05:28
Log: sessions.txt
IoSession with /111.19.92.156:18194 opened on 11-07-2018 05:41
Log: sessions.txt
IoSession with /111.19.92.156:18323 opened on 11-07-2018 05:49
Log: sessions.txt
IoSession with /111.19.92.156:3977 opened on 11-07-2018 07:37
Log: sessions.txt
IoSession with /111.19.92.156:3969 opened on 11-07-2018 07:41
Log: sessions.txt
IoSession with /111.19.92.156:4000 opened on 11-07-2018 07:42
Log: sessions.txt
IoSession with /119.28.178.241:58638 opened on 11-07-2018 10:08
HeavenMS v83 starting up.

Skills loaded in 1.625 seconds
Items loaded in 0.533 seconds
Quest loaded in 2.495 seconds

Starting world 0
    Channel 1: Listening on port 7575
    Channel 2: Listening on port 7576
    Channel 3: Listening on port 7577
Finished loading world 0

Listening on port 8484


HeavenMS is now online.

```

All these `IoSession with /xxx.xxx.xxx.xxx:xxxxx opened on` should be connection log from login server. However can't get any log from channel server after crashed. During the crashing, the hourly auto-save does not work, either.

**Comment by xinyifly** [comment]
This is another crash log:
```
Log: saveCharAuto.txt
Attempting to autosave Tommm at Tue Jul 10 16:09:33 GMT 2018
Log: saveCharAuto.txt
Attempting to autosave Nate at Tue Jul 10 16:09:33 GMT 2018
Log: saveCharAuto.txt
Attempting to autosave phphp at Tue Jul 10 16:09:34 GMT 2018
Log: saveCharAuto.txt
Attempting to autosave Forest at Tue Jul 10 16:09:34 GMT 2018
Log: saveChar.txt
Attempting to save Tommm at Tue Jul 10 16:09:52 GMT 2018
Log: sessions.txt
IoSession with /119.28.178.241:35858 opened on 10-07-2018 16:09
Error thrown: PacketHandler/net.server.channel.handlers.ItemPickupHandler.txt
java.lang.NullPointerException
	at net.server.channel.handlers.ItemPickupHandler.handlePacket(ItemPickupHandler.java:49)
	at net.MapleServerHandler.messageReceived(MapleServerHandler.java:166)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain$TailFilter.messageReceived(DefaultIoFilterChain.java:690)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain.callNextMessageReceived(DefaultIoFilterChain.java:417)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain.access$1200(DefaultIoFilterChain.java:47)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain$EntryImpl$1.messageReceived(DefaultIoFilterChain.java:765)
	at org.apache.mina.filter.codec.ProtocolCodecFilter$ProtocolDecoderOutputImpl.flush(ProtocolCodecFilter.java:407)
	at org.apache.mina.filter.codec.ProtocolCodecFilter.messageReceived(ProtocolCodecFilter.java:236)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain.callNextMessageReceived(DefaultIoFilterChain.java:417)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain.access$1200(DefaultIoFilterChain.java:47)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain$EntryImpl$1.messageReceived(DefaultIoFilterChain.java:765)
	at org.apache.mina.core.filterchain.IoFilterAdapter.messageReceived(IoFilterAdapter.java:109)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain.callNextMessageReceived(DefaultIoFilterChain.java:417)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain.fireMessageReceived(DefaultIoFilterChain.java:410)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.read(AbstractPollingIoProcessor.java:710)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.process(AbstractPollingIoProcessor.java:664)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.process(AbstractPollingIoProcessor.java:653)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.access$600(AbstractPollingIoProcessor.java:67)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor$Processor.run(AbstractPollingIoProcessor.java:1124)
	at org.apache.mina.util.NamePreservingRunnable.run(NamePreservingRunnable.java:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
	at java.lang.Thread.run(Thread.java:748)

Error thrown: PacketHandler/net.server.channel.handlers.ItemPickupHandler.txt
java.lang.NullPointerException
	at net.server.channel.handlers.ItemPickupHandler.handlePacket(ItemPickupHandler.java:49)
	at net.MapleServerHandler.messageReceived(MapleServerHandler.java:166)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain$TailFilter.messageReceived(DefaultIoFilterChain.java:690)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain.callNextMessageReceived(DefaultIoFilterChain.java:417)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain.access$1200(DefaultIoFilterChain.java:47)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain$EntryImpl$1.messageReceived(DefaultIoFilterChain.java:765)
	at org.apache.mina.filter.codec.ProtocolCodecFilter$ProtocolDecoderOutputImpl.flush(ProtocolCodecFilter.java:407)
	at org.apache.mina.filter.codec.ProtocolCodecFilter.messageReceived(ProtocolCodecFilter.java:236)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain.callNextMessageReceived(DefaultIoFilterChain.java:417)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain.access$1200(DefaultIoFilterChain.java:47)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain$EntryImpl$1.messageReceived(DefaultIoFilterChain.java:765)
	at org.apache.mina.core.filterchain.IoFilterAdapter.messageReceived(IoFilterAdapter.java:109)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain.callNextMessageReceived(DefaultIoFilterChain.java:417)
	at org.apache.mina.core.filterchain.DefaultIoFilterChain.fireMessageReceived(DefaultIoFilterChain.java:410)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.read(AbstractPollingIoProcessor.java:710)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.process(AbstractPollingIoProcessor.java:664)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.process(AbstractPollingIoProcessor.java:653)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.access$600(AbstractPollingIoProcessor.java:67)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor$Processor.run(AbstractPollingIoProcessor.java:1124)
	at org.apache.mina.util.NamePreservingRunnable.run(NamePreservingRunnable.java:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
	at java.lang.Thread.run(Thread.java:748)

Log: usedCommands.txt
Nate used: @int 5 on 10-07-2018 16:16

Log: saveChar.txt
Attempting to save Nate at Tue Jul 10 16:17:13 GMT 2018
Log: saveChar.txt
Attempting to save Nate at Tue Jul 10 16:17:25 GMT 2018
Log: usedCommands.txt
Forest used: !warp 103000105 on 10-07-2018 16:19

Log: usedCommands.txt
Forest used: !warp 200000121 on 10-07-2018 16:19

Log: saveChar.txt
Attempting to save Forest at Tue Jul 10 16:20:02 GMT 2018
disconnect by exception
java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:384)
	at org.apache.mina.transport.socket.nio.NioProcessor.read(NioProcessor.java:273)
	at org.apache.mina.transport.socket.nio.NioProcessor.read(NioProcessor.java:44)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.read(AbstractPollingIoProcessor.java:690)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.process(AbstractPollingIoProcessor.java:664)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.process(AbstractPollingIoProcessor.java:653)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor.access$600(AbstractPollingIoProcessor.java:67)
	at org.apache.mina.core.polling.AbstractPollingIoProcessor$Processor.run(AbstractPollingIoProcessor.java:1124)
	at org.apache.mina.util.NamePreservingRunnable.run(NamePreservingRunnable.java:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
	at java.lang.Thread.run(Thread.java:748)
Log: saveChar.txt
Attempting to save phphp at Tue Jul 10 16:20:02 GMT 2018
Log: sessions.txt
IoSession with /119.28.178.241:35938 opened on 10-07-2018 16:23
Log: sessions.txt
IoSession with /119.28.178.241:35940 opened on 10-07-2018 16:23
Log: usedCommands.txt
Forest used: !warp 200090100 on 10-07-2018 16:25

Log: usedCommands.txt
Forest used: !hide on 10-07-2018 16:25

Log: usedCommands.txt
Forest used: !hide on 10-07-2018 16:25

Log: usedCommands.txt
Forest used: !hide on 10-07-2018 16:26

Log: usedCommands.txt
Forest used: !hide on 10-07-2018 16:26

Log: usedCommands.txt
Forest used: !hide on 10-07-2018 16:26

Log: usedCommands.txt
Forest used: !hide on 10-07-2018 16:29

Log: saveChar.txt
Attempting to save phphp at Tue Jul 10 16:30:00 GMT 2018
Log: saveChar.txt
Attempting to save phphp at Tue Jul 10 16:30:10 GMT 2018
Log: usedCommands.txt
Forest used: !warp 300030100 on 10-07-2018 16:34

Log: usedCommands.txt
Forest used: !hide on 10-07-2018 16:34

Log: usedCommands.txt
Forest used: !help on 10-07-2018 16:36

Log: sessions.txt
IoSession with /111.19.92.156:18597 opened on 10-07-2018 16:36
Log: usedCommands.txt
Forest used: !warphere human on 10-07-2018 16:38

Log: usedCommands.txt
Forest used: !setgmlevel Human 2 on 10-07-2018 16:38

Log: usedCommands.txt
Forest used: !help on 10-07-2018 16:39

Log: usedCommands.txt
Forest used: @help on 10-07-2018 16:40

Log: usedCommands.txt
Forest used: !help on 10-07-2018 16:40

Log: usedCommands.txt
Human used: !level 45 on 10-07-2018 16:41

Log: usedCommands.txt
Forest used: !warphere human on 10-07-2018 16:43

Log: usedCommands.txt
Forest used: !job 242 on 10-07-2018 16:45

Log: usedCommands.txt
Forest used: !job 232 on 10-07-2018 16:46

Log: usedCommands.txt
Forest used: !job 900 on 10-07-2018 16:46

Log: usedCommands.txt
Human used: !maxstat on 10-07-2018 16:47

Log: sessions.txt
IoSession with /111.19.92.156:26282 opened on 10-07-2018 16:48
Log: sessions.txt
IoSession with /111.19.92.156:26289 opened on 10-07-2018 16:48
Log: sessions.txt
IoSession with /119.28.178.241:36656 opened on 10-07-2018 16:50
Log: sessions.txt
IoSession with /119.28.178.241:36658 opened on 10-07-2018 16:50
Log: sessions.txt
IoSession with /119.28.178.241:36706 opened on 10-07-2018 16:51
Log: sessions.txt
IoSession with /119.28.178.241:36960 opened on 10-07-2018 16:52
Log: sessions.txt
IoSession with /119.28.178.241:36988 opened on 10-07-2018 16:53
Log: sessions.txt
IoSession with /119.28.178.241:36990 opened on 10-07-2018 16:53
Log: sessions.txt
IoSession with /119.28.178.241:37026 opened on 10-07-2018 16:53
Log: sessions.txt
IoSession with /119.28.178.241:37050 opened on 10-07-2018 16:54
Log: sessions.txt
IoSession with /119.28.178.241:37052 opened on 10-07-2018 16:54
Log: sessions.txt
IoSession with /119.28.178.241:37070 opened on 10-07-2018 16:55
Log: sessions.txt
IoSession with /111.19.92.156:22692 opened on 10-07-2018 16:55
Log: sessions.txt
IoSession with /111.19.92.156:22700 opened on 10-07-2018 16:55
Log: sessions.txt
IoSession with /119.28.178.241:37104 opened on 10-07-2018 16:56
Log: sessions.txt
IoSession with /111.19.92.156:18599 opened on 10-07-2018 16:56
Log: sessions.txt
IoSession with /119.28.178.241:37200 opened on 10-07-2018 16:59
Log: sessions.txt
IoSession with /119.28.178.241:37354 opened on 10-07-2018 17:00
Log: sessions.txt
IoSession with /111.19.92.156:18577 opened on 10-07-2018 17:01
Log: sessions.txt
IoSession with /119.28.178.241:37374 opened on 10-07-2018 17:01
Log: sessions.txt
IoSession with /47.75.123.91:25612 opened on 10-07-2018 17:11
Log: sessions.txt
IoSession with /47.75.123.91:28394 opened on 10-07-2018 17:14
Log: sessions.txt
IoSession with /172.17.0.1:44870 opened on 10-07-2018 17:32
Log: sessions.txt
IoSession with /111.19.92.156:22708 opened on 10-07-2018 17:34
HeavenMS v83 starting up.

Skills loaded in 1.727 seconds
Items loaded in 0.594 seconds
Quest loaded in 2.28 seconds

Starting world 0
    Channel 1: Listening on port 7575
    Channel 2: Listening on port 7576
    Channel 3: Listening on port 7577
Finished loading world 0

Listening on port 8484


HeavenMS is now online.

```

**Comment by ronancpl** [comment]
Issue merged with #316.
